### ISSUE TITLE
Dungeon Master 2 - Loading a save freezes the program up

### ISSUE THREAD

**Issue by MechaGodJovi** [dosbox-staging/dosbox-staging#695]
Saving/loading works fine up until a certain point which is relatively early in the game that upon trying to load a saved game after quitting and coming back it freezes the program with no error and just hangs. To reproduce the problem it seems to only happen when I go to the area with the wolves for the 2nd clan piece. 

If I save anytime after that and quit the program then come back later and try to load ANY files for that game that's tied to my current party then it freezes. By that I mean it doesn't matter if I had a backup 30 minutes ago or a save at the start of that game, it just won't work unless I start a new one. Game works fine as long as I keep it running so the game works but somethings broken. It also happens on the regular DOSBox 0.74-3 so unsure if it's something with my version of the game or the program itself.

Windows 10
Intel n3150 1.6GHz, Intel HD Graphics 3000
DOSBox-Staging Version 0.75.2

For DOSBox 0.74-3 the only thing I've done with it was change the resolution to match desktop and changed output to openglnb. For DOSBox-Staging Version 0.75.2 everything was kept as default.

**Comment by kcgen** [comment]
Thanks for the report @MechaGodJovi.

I'm not able to reproduce this, however I haven't made it to the area with wolves. To help speed that up, would you be able to zip and attach your save games?

For reference, my MD5 checksums of the notable executable and data files are:

**md5sum skull.exe intro end  ftl credits data/graphics.dat** 

``` text
c03ec00e0f4e3d9b7fca6c26ced5936f  skull.exe
8bca0cc92d0839dd47962a0d54b4f783  intro
ba2f2875963c47bd4195bf0a041d702d  end
a354faedff81662cc8646a10acd94a4e  ftl
8d83a751db963182d66efe6ca8af1c8b  credits
25247ede4dabb6a71e5dabdfbcd5907d  data/graphics.dat
```

**Comment by MechaGodJovi** [comment]
Sorry I had to reinstall my OS and reformat after some issues this morning that caused my system to freeze during startup after my kid forcefully tried turning it off and on a few times during an update.

I don't have the saves now but I can try to reproduce this and upload them. Shouldn't take very long it's the 2nd time it's happened on two seperate new games but seems to always happen after that point.

Am I supposed to do something with the md5 checksum? I've been on Github before but never really reported issues till someone lead me this way from reddit so I don't know what I need to do with that information sorry!

**Comment by FeignInt** [comment]
Hi @MechaGodJovi 

Regards the md5sums, I believe the intent was to confirm that game was using the same file versions. to discount it being an issue with the game files, file corruption or different (buggy) version.

md5sum is like a fingerprint "unique" to the file, it is not perfect but for most purposes it is good enough

I'm not a windows person.. so did a quick web search and found this
https://docs.microsoft.com/en-us/troubleshoot/windows-client/deployment/compute-md5sha-1-cryptographic-hash-value

so something like
```
FCIV -md5 skull.exe
```
and compare the returned md5sum


If on Linux ( or linux like )
assuming you are in "the game directory" 
```shell
md5sum -c <<EOF
c03ec00e0f4e3d9b7fca6c26ced5936f  skull.exe
<snip>..aphics.dat
EOF
```
would check the md5sums match the provided list (easier than eyeballing them)

You may be able to do something similar on Windows with FCIV or some other utility, or at least list multiple files for the one FCIV command.

Hope that helps

**Comment by kcgen** [comment]
Thanks for the helpful response @FeignInt.
Yup, reducing variables was the intent of the checksums: simply determine if we're working with the same version of the game (or not).

The fewer variables the better :-)

**Comment by kcgen** [comment]
@MechaGodJovi , I think 7zip installs a checksum tool too, if you have context menus enabled. 

In Windows Explorer, try right-clicking on a file (or files) and see a 7zip checksum menu is available: maybe it will say SHA1 or SHA256. 

If you can get either of those sets of values, I can compare them again mine too.

**Comment by MechaGodJovi** [comment]
**@FeignInt** Ah ok thanks that helps. 

c03ec00e0f4e3d9b7fca6c26ced5936f skull.exe
26cb6b99bd43362d81f9c54a841e9b4e intro
a897ccb793af57ee0a8de43346d9f05f ftl
7c3f291bf24e3517e79cd779f40b1308 end
2cdd656cf00af8585bbd5f0e4af1aa63 credits.pcx
25247ede4dabb6a71e5dabdfbcd5907d graphics.dat

Well I got the 3rd clan key and saves are still working fine oddly enough despite doing what I did before but don't wanna jinx it though. I started another new game because I'm still determined to find the issue but kind of wondering if maybe it was just a game bug or something.

**@kcgen** 

I'm trying my best to reproduce it but what seemed to happen back to back before isn't happening now and it's kind of frustrating me now. Yeah that is one thing I hated about Microsoft and this computer is junk so I just use it for old games anyway so not like I lost anything important. I still feel bad having to go through all that without being able to provide the files though and felt like I wasted your guys time but I'm still gonna try to reproduce the issue, it's the least I can do.

Also did not know 7zip did that so that's good to know and sounds like an easier way to do it.

**Comment by FeignInt** [comment]
@MechaGodJovi 

The *skull.exe* and *data/graphics.dat* appear to be the same, but the other files differ..
Not indicative of a problem, just something to keep in mind that the game files are different.

Since you have not been able to reproduce the problem and given the history of  "loading a save freezes" and a "stalled?.. Windows Update" I would lean towards a failing HDD being the culprit.

This is only an assumption, based on the fault not being reproducible.

If it is a failing HDD I suspect the the OS reinstall is using less space and as such not trying to use the failing portion of the HDD yet.

Is it an old HDD ?

I imagine there is some built in tool on Windows10 for checking HDD health
[quick search got me this](https://social.technet.microsoft.com/wiki/contents/articles/53235.windows-10-built-in-tools-for-hard-disk-health-check.aspx)

There may be better tools for the job, I will defer to anyone with current Windows experience.

**Comment by kcgen** [comment]
Interesting @MechaGodJovi and @FeignInt ... maybe a fresh install (to different healthier sectors) has indirectly "solved" it. Great if so!

**Comment by MechaGodJovi** [comment]
**@FeignInt @kcgen**  It's about 6 years old so definitely had it for awhile. I ran chkdsk and ran a command through cmd but everything came back okay as far as I can tell.

Computer has been slowing down lately for sure so I guess the HDD failing wouldn't be out of the question too since I have been heavily using it on a daily basis mostly for work related things.

I've gone through the beginning of the game up till the 2nd clan piece twice since last night with no issues and continued on with my current one no problem so I really don't know. Should I close this or? Well again sorry I feel like I wasted peoples time for an issue that just resolved itself but I guess if it happens again I know where to come back too and I'll make sure to make a backup of the saves next time if it by chance does.
